packages/storybook/src/generators/migrate-9/helper-functions.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  generateFiles,
+  getPackageManagerCommand,
+  output,
+  Tree,
+  workspaceRoot,
+  visitNotIgnoredFiles,
+  joinPathFragments,
+  readJson,
+  detectPackageManager,
+} from '@nx/devkit';
+import { fileExists } from 'nx/src/utils/fileutils';
+import { readFileSync } from 'fs';
+import { dirname, join } from 'path';
+
+export function onlyShowGuide(storybookProjects: {
+  [key: string]: {
+    configDir: string;
+  };
+}) {
+  const packageManager = detectPackageManager();
+  const pm = getPackageManagerCommand(packageManager);
+
+  output.log({
+    title: 'Storybook 8 Migration Guide',
+    bodyLines: [
+      `You can run the following commands manually to upgrade your Storybook projects to Storybook 8:`,
+      ``,
+      `1. Call the Storybook upgrade script:`,
+      `${pm.exec} ${
+        packageManager === 'yarn' ? 'storybook' : 'storybook@latest'
+      } upgrade`,
+      ``,
+      `2. Call the Storybook automigrate scripts:`,
+      `Run the following commands for each Storybook project:`,
+      ...Object.entries(storybookProjects).map(
+        ([_projectName, storybookProjectInfo]) => {
+          return `${pm.exec} ${
+            packageManager === 'yarn' ? 'storybook' : 'storybook@latest'
+          } automigrate --config-dir ${storybookProjectInfo.configDir}`;
```

Comment:
for the upgrade command we normally as for users to run it versioned e.g.  `npx storybook@<some-version> upgrade` but for automigrations and other commands we normally want the user's installed binary to be used